src/managers/AuditLogManager.ts

```diff
@@ -0,0 +1,97 @@
+import { EventEmitter } from '@tauri-apps/api/shell';
+
+export type RequestEvent = {
+	timestamp: Date;
+	chronology: 'before' | 'after';
+	eventType: `${'pre' | 'post'}${'Service' | 'Endpoint' | 'Request'}Script` | 'request' | 'root';
+	associatedId?: string;
+	error?: string;
+};
+
+export type AuditLog = RequestEvent[];
+
+export type AuditUpdateEvent = 'update';
+
+export type TransformedAuditLog = {
+	before: RequestEvent;
+	after: RequestEvent;
+	innerEvents: TransformedAuditLog[];
+};
+
+type AuditEventEmitterListenerType = {
+	update: (newEvent: RequestEvent) => void;
+};
+
+export class AuditLogManager extends EventEmitter<AuditUpdateEvent> {
+	public static INSTANCE = new AuditLogManager();
+
+	private constructor() {
+		super();
+	}
+
+	addToAuditLog(
+		auditLog: AuditLog,
+		chronology: RequestEvent['chronology'],
+		eventType: RequestEvent['eventType'],
+		associatedId?: string,
+		error?: string,
+	) {
+		const newRequestEvent: RequestEvent = {
+			timestamp: new Date(),
+			chronology,
+			eventType,
+			error,
+			associatedId,
+		};
+		auditLog.push(newRequestEvent);
+		this.emit('update', newRequestEvent);
+	}
+
+	getEventDataType(event: RequestEvent) {
+		const dataTypes = ['Service', 'Endpoint', 'Request', 'request'] as const;
```

Comment:
What's the difference between a request and a Request, and is there any way to normalize those?

Reply:
Since we're refactoring this it doesn't really matter, but this is `preRequestService` vs `request` type difference